src/backend/access/heap/rewriteheap.c

```diff
@@ -1173,7 +1138,7 @@ heap_xlog_logical_rewrite(XLogReaderState *r)
 				 errmsg("could not fsync file \"%s\": %m", path)));
 	pgstat_report_wait_end();
 
-	wallog_mapping_file(path, fd);
+	wallog_file_descriptor(path, fd, -1);
```

Comment:
This is a WAL redo function. It has no business creating new WAL records.

Reply:
Fixed